src/main/java/duke/storage/TaskList.java

```diff
@@ -0,0 +1,93 @@
+package duke.storage;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import duke.exception.DukeException;
+import duke.task.Task;
+
+/**
+ * Represents a Task list that can be added to and removed from.
+ */
+public class TaskList {
+    /* The current list of tasks */
+    private List<Task> tasks;
+
+    /**
+     * Creates a task list with a given initial list of tasks.
+     *
+     * @param tasks The initial list of tasks.
+     */
+    public TaskList(List<Task> tasks) {
+        this.tasks = tasks;
+    }
+
+    /**
+     * Adds a task to the existing list of tasks.
+     *
+     * @param task The task to be added.
+     */
+    public void add(Task task) {
+        this.tasks.add(task);
+    }
+
+    /**
+     * Removes a task by its serial number from the list. Returns the task that was removed.
+     *
+     * @param serialNo The serial number of the task to remove.
+     * @return The task that was removed.
+     */
+    public Task remove(int serialNo) throws DukeException {
+        if (tasks.isEmpty()) {
+            throw new DukeException("Sorry Boss, there is no task to remove.");
+        } else if (serialNo < 1 || serialNo > tasks.size()) {
+            throw new DukeException("Sorry Boss, please provide the correct serial no.");
+        }
+        return this.tasks.remove(serialNo - 1);
+    }
```

Comment:
Just a minor suggestion, considering "delete" would be the input for this method to be called, maybe using delete as method name and printed output would be better?